src/redux/thunks/file.ts

```diff
@@ -1154,6 +1156,66 @@ export function batchGetDirectLinks(index: number, files: FileResponse[]): AppTh
   };
 }
 
+export function resetThumbnails(files: FileResponse[]): AppThunk {
+  return async (dispatch, _getState) => {
+    const cache = getCachedThumbExts();
+    const uris = files
+      .filter((f) => f.type == FileType.file)
+      .filter((f) =>
+        cache === undefined || cache === null ? true : cache.has((fileExtension(f.name) || "").toLowerCase()),
+      )
+      .map((f) => getFileLinkedUri(f));
+
+    if (uris.length === 0) {
+      enqueueSnackbar({
+        message: i18next.t("application:fileManager.noFileCanResetThumbnail"),
+        preventDuplicate: true,
+        variant: "warning",
+        action: DefaultCloseAction,
+      });
+      return;
+    }
+
+    try {
+      await dispatch(
+        sendResetFileThumbs({
+          uris,
+        }),
+      );
+
+      enqueueSnackbar({
+        message: i18next.t("application:fileManager.resetThumbnailRequested"),
+        variant: "success",
+        action: DefaultCloseAction,
+      });
+    } catch (_e) {
+      // Error snackbar is handled in send()
+    } finally {
+      // Clear cached thumbnails so they will be reloaded next time
+      files
+        .filter((f) => f.type == FileType.file)
+        .forEach((f) =>
+          dispatch(
+            fileUpdated({
```

Comment:
I don't recall `fileUpdated` will clear thumbnail cache. We need a new method to delete item from:
https://github.com/MasonDye/frontend/blob/77ebb6d90b65bc8eb3eaa78dccad5ea39a619637/src/redux/fileManagerSlice.ts#L85C3-L85C8

Reply:
Oh never mind, it is clearing thumb cache.